### ISSUE TITLE
issue w/ parsing remarks

### ISSUE THREAD

**Issue by BradWalker** [mivek/MetarParser#256]
**Describe the bug**
The remarks section in the  following METAR is not fully parsed.

"KCOS 261454Z 34012G18KT 10SM BKN060 BKN100 15/10 A3020 RMK AO2 RAE45 SLP171 P0000 60002 T01500100 51009 $"

**To Reproduce**
```
        MetarService service = MetarService.getInstance();

        Metar metar = null;

        try {
            metar = service.decode(code);
        } catch (ParseException ex) {
            Logger.getLogger(JavaApplication1.class.getName()).log(Level.SEVERE, null, ex);
        }
 
        System.out.println(metar);
```

**Expected behavior**
MetarParser is pretty awesome.. I would have thought it would fully parse the remarks section.

**Screenshots**
Here is what I get:
`io.github.mivek.model.Metar@2e5d6d97[day of the month=26,time of report=14:54,airport=City of Colorado Springs Municipal Airport (KCOS),speed=12,unit=KT,direction=North North West,direction (degrees)=340,gusts=18,minimal wind variation=0,maximal wind variation=0,main visibility=10SM,minimum visibility=0,minimum visibility direction=<null>,vertical visibility (ft)=<null>,clouds=[io.github.mivek.model.Cloud@5910e440[quantity=broken,type=<null>,height (ft)=6000], io.github.mivek.model.Cloud@533ddba[quantity=broken,type=<null>,height (ft)=10000]],weather conditions=[],cavok=false,remarks=automated station with a precipitation discriminator RAE45 sea level pressure of 1017.1 HPa P0000 60002 T01500100 51009 $ ,original message=KCOS 261454Z 34012G18KT 10SM BKN060 BKN100 15/10 A3020 RMK AO2 RAE45 SLP171 P0000 60002 T01500100 51009 $,temperature (Â°C)=15,dew point=10,altimeter (hPa)=1022,nosig=false,auto=false,runways information=[],trends=[]]`

**Desktop (please complete the following information):**
n/a

**Smartphone (please complete the following information):**
n/a

**Additional context**
none

**Comment by mivek** [comment]
Hello @BradWalker 

Remarks parsing is based on the federal [meteorological handbook](https://www.icams-portal.gov/publications/fmh/FMH1/FMH1.pdf). Most of them are implemented in the parser.

Regarding the code "KCOS 261454Z 34012G18KT 10SM BKN060 BKN100 15/10 A3020 RMK AO2 RAE45 SLP171 P0000 60002 T01500100 51009 $"
`P0000` and  `60002` are about precipitations amount and will be implemented in remark parsing in future versions.

However I do not know what `RAE45`, `T01500100 ` or `51009 ` mean.

**Comment by BradWalker** [comment]
@mivek ,

Here is the decoding for this remark:
`RMK AO2 RAE45 SLP171 P0000 60002 T01500100 51009 $`

AO2 - station type
RAE45 - rain ending, sec. 12.7.1(k) of the meteorological handbook
SLP171 - sea-level pressure 917.1 hectopascals, sec. 12.7.1(w)
P0000 60002 - precipitation related, you already know mentioned this
T01500100 - hourly temp & dew point, sec. 12.7.2(d)
51009 - 3-hourly pressure tendency, sec. 12.7.2(h)

BTW, I really, really like the engineering work you put into this. It's nice. If I had to do this, I would make a complete mess of it.

**Comment by mivek** [comment]
Hello @BradWalker 

Thanks for the feed back.

I finished the implementation of the remarks. Here is the remark "automated station with a precipitation discriminator  rain ending at :45 sea level pressure of 1017.1 HPa 0/100 of an inch of precipitation fell in the last hour 0.02 inches of precipitation fell in the last 6 hours hourly temperature of 15°C and dew point of 10°C Increase, then steady, or increase then Increase more slowly of 0.9 hectopascals in the past 3 hours $".

I will add commas between each remark to make the string more readable.

**Comment by BradWalker** [comment]
> Hello @BradWalker
> 
> Thanks for the feed back.
> 
> I finished the implementation of the remarks. 
> 
> I will add commas between each remark to make the string more readable.

Hey @mivek , I'm grateful for your prompt efforts to fix this.. The code is a real nice piece of work.
So I bought you a Starbucks giftcard. Just to let you know that people out here in nowhere land do recognize your open source efforts.
Should be in your email.

-brad w.

**Comment by mivek** [comment]
Hello @BradWalker 

Thanks for the gift, I received the email. The fix is available in version 2.3.1 on [maven central](https://search.maven.org/artifact/io.github.mivek/metarParser/2.3.1/pom).
